### PR TITLE
#526: Each main component should export its `props` (closes #526)

### LINKED ISSUE #526: Each main component should export its `props`
## requirements
In our projects we often need to wrap a brc component in order to add some project-custom features/defaults to it.
In some of these cases we also add/edit some proptypes definitions.
to avoid setting `@props` to `{ strict: false }` every brc component should export its own proptypes:

```js
// BAD
@props({
  newCustomProp: t.String
}, { strict: false })

// NICE
import Component, { Props } from 'buildo-react-components/lib/Component';

@props({
  ...Props,
  newCustomProp: t.String
})
```

## specs
- move proptypes in external variable `Props`
- export `Props`

```js
export const Props = {...};

@props(Props)
```

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by veej**
Issue #526

### tests performed
Everything should work as before (i.e. every component page in showroom should work)

### CODE REVIEW

File: src/transition-wrapper/TransitionWrapper.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const PropTypes = {
   style: t.maybe(t.Object)
 };
 
-@props(PropTypes, { strict: false })
+@props(Props)
```
Comment (FrancescoCioria) [inline src/transition-wrapper/TransitionWrapper.js]: you accidentally removed `{strict: false}`

File: src/dropdown/Dropdown.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ class Select extends _Select {
  * @param flat - whether it should have a flat style
  */
 @skinnable()
-@props(PropTypes, { strict: false })
```
Comment (FrancescoCioria) [inline src/dropdown/Dropdown.js]: you accidentally removed `{strict: false}`

File: src/flex/FlexView.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ const PropTypes = {
  * @param height - height property (for parent secondary axis)
  * @param width - width property (for parent secondary axis)
  */
-@props(PropTypes, { strict: false })
```
Comment (FrancescoCioria) [inline src/flex/FlexView.js]: you accidentally removed `{strict: false}`

File: src/focusable-view/FocusableView.js [hunk 4]
```diff
@@ -11,16 +22,7 @@ import { props, t, skinnable } from '../utils';
...
-@props({
-  children: t.union([t.ReactChildren, t.Function]),
-  onFocus: t.maybe(t.Function),
-  onBlur: t.maybe(t.Function),
-  tabIndex: t.maybe(t.Number),
-  component: t.maybe(t.union([t.Function, t.String])),
-  ignoreFocus: t.maybe(t.Boolean),
-  className: t.maybe(t.String),
-  style: t.maybe(t.Object)
-}, { strict: false })
```
Comment (FrancescoCioria) [inline src/focusable-view/FocusableView.js]: you accidentally removed `{strict: false}`

File: src/scroll/ScrollView.js [hunk 5]
```diff
@@ -38,7 +38,7 @@ const PropTypes = {
   style: t.maybe(t.Object)
 };
 
-@props(PropTypes, { strict: false })
```
Comment (FrancescoCioria) [inline src/scroll/ScrollView.js]: you accidentally removed `{strict: false}`

File: src/text-overflow/TextOverflow.js [hunk 6]
```diff
@@ -7,22 +7,24 @@ import { warn } from '../utils/log';
...
+};
+
+/**
+ * Text view which, if string content is too large, trims it and shows the full content on "hover" with a `Popover` (or custom component if any)
+ * @param children - in case you want to use a custom component (like a `Tooltip`) to render the full content which is passed as the first argument
+ * @param label - this is the full string
+ * @param lazy - whether the tooltip appearance should be delayed after mouse entering or not
+ * @param delayWhenLazy - tooltip delay if the component is lazy
+ */
+@props(Props)
```
Comment (FrancescoCioria) [inline src/text-overflow/TextOverflow.js]: you accidentally removed `{strict: false}`

File: src/confirmation-input/ConfirmationInput.js [hunk 7]
```diff
@@ -36,7 +25,20 @@ import Icon from '../Icon/Icon';
   className: t.maybe(t.String),
   id: t.maybe(t.String),
   style: t.maybe(t.Object)
-}, { strict: false })
```
Comment (FrancescoCioria) [inline src/confirmation-input/ConfirmationInput.js]: you accidentally removed `{strict: false}`

**Comment by FrancescoCioria** [comment]
@veej you accidentally removed `{strict: false}`s 😜

**Comment by veej** [comment]
Is it not the purpose of the PR to avoid `{strict: false}`s? I supposed we would handle those cases by redefining `Props` with additional values.

**Comment by FrancescoCioria** [comment]
👍 💯

**Pull request merged**